polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -1194,6 +1207,34 @@ impl<T: Config> QueryHandler for Pallet<T> {
 }
 
 impl<T: Config> Pallet<T> {
+	/// Validate `assets` to be reserve-transferred and return their reserve location.
+	fn validate_assets_and_find_reserve(
+		assets: &[MultiAsset],
+		dest: &MultiLocation,
+	) -> Result<TransferType, Error<T>> {
+		let mut reserve = None;
+		for asset in assets.iter() {
+			if let Fungible(x) = asset.fun {
+				// If fungible asset, ensure non-zero amount.
+				ensure!(!x.is_zero(), Error::<T>::Empty);
+			}
+			let transfer_type =
+				<T::XcmExecutor as AssetTransferSupport>::determine_for(&asset, dest)
```

Comment:
```suggestion
				T::XcmExecutor::determine_for(&asset, dest)
```

it works without `<T::XcmExecutor as AssetTransferSupport>`, I dont know if we want to be explicit here